crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -195,6 +210,10 @@ where
 
         transaction.commit()?;
 
+        let current_stf_version = block.header().state_transition_bytecode_version();
+        self.stf_parameters_provider
+            .cache_stf_version(current_stf_version);
```

Comment:
We can have this login inside of the `ConsensusParametersProvider`. We don't need to teach off-chain worker to provide this information=)

<img width="1421" alt="image" src="https://github.com/user-attachments/assets/7fe8ed8f-318b-4f9d-a232-7d371f059231" />

Reply:
You're right, that was a bit of a code smell since the beginning.
Updated in 932a1c55506b548fc60ed0527b979559cbdf9124